### PR TITLE
Remove dependency on `glog`

### PR DESCRIPTION
**Pull request by tamird**
Addresses part of #217.

### CODE REVIEW

**Comment by iamqizhao** [comment]
glog is the default logger in the library...

**Comment by tamird** [comment]
Right, but this is a problem for folks who have e.g. forked glog. This change promotes `StdLogger` to the default logger. Users should have no problem using glog if they so choose.

**Comment by xiang90** [comment]
@iamqizhao Why do we use glog as the default logger? For performance reason? As https://github.com/grpc/grpc-go/issues/217 said, it registers some unwanted flags.

**Comment by iamqizhao** [comment]
I think it is a good and convenient choice for most of users ...

**Comment by tamird** [comment]
It is a problem for *some* users and is impossible to work around.

**Comment by iamqizhao** [comment]
On Mon, Jun 8, 2015 at 11:43 AM, Tamir Duberstein <notifications@github.com>
wrote:

> It is a problem for *some* users and is impossible to work around.
>
Why? You can set whatever logger you want.

**Comment by iamqizhao** [comment]
On Mon, Jun 8, 2015 at 11:47 AM, Qi Zhao <toqizhao@gmail.com> wrote:

>
>
> On Mon, Jun 8, 2015 at 11:43 AM, Tamir Duberstein <
> notifications@github.com> wrote:
>
>> It is a problem for *some* users and is impossible to work around.
>>
> Why? You can set whatever logger you want.
>
oh, you meant no way to avoid flag pollution ... hmm...

**Comment by xiang90** [comment]
@iamqizhao Yes. 

Take a look at
https://github.com/golang/glog/blob/master/glog.go#L398-L411

**Comment by iamqizhao** [comment]
I got the points from you guys. Let me think about it.

File: grpclog/logger.go [hunk 1]
```diff
@@ -39,19 +39,14 @@ package grpclog // import "google.golang.org/grpc/grpclog"
...
-	// GLogger is a Logger that uses glog. This is the default logger.
-	GLogger Logger = &glogger{}
-
-	// StdLogger is a Logger that uses golang's standard logger.
-	StdLogger Logger = log.New(os.Stderr, "", log.LstdFlags)
+var logger Logger
 
-	logger = GLogger
-)
+func init() {
```
Comment (iamqizhao) [inline grpclog/logger.go]: I would prefer a variable initialization instead of init().

var logger = log.New(os.Stderr, "", log.LstdFlags)

Comment (tamird) [inline grpclog/logger.go]: Done

**Pull request merged**